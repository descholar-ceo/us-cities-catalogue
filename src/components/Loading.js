const Loading = () => (<div className="loader" />);

export default Loading;
