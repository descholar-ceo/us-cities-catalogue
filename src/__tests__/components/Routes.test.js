import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Loading from '../../components/Loading';

describe('Test Loading component', () => {
  it('Renders the loading component', done => {
    const loadingComp = shallow(<Loading />);
    expect(loadingComp.type()).toBe('div');
    expect(loadingComp.find('div.loader').exists()).toBe(true);
    done();
  });
});

describe('Test the snapshot of Loading component', () => {
  it('Render the correct snapshot', done => {
    const loadingSnapshot = renderer.create(<Loading />).toJSON();
    expect(loadingSnapshot).toMatchSnapshot();
    done();
  });
});
