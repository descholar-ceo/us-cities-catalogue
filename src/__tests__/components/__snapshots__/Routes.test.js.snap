// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the snapshot of Loading component Render the correct snapshot 1`] = `
<div
  className="loader"
>
  <h1
    className="loading-text"
  >
    Loading...
  </h1>
</div>
`;
